core/src/main/java/hudson/model/Job.java

```diff
@@ -1070,6 +1070,57 @@ public PermalinkList getPermalinks() {
     }
 
     /**
+     * The following methods should be overriden by Jobs that support being disabled:
+     * - supportsDisable() - override to return true
+     * - isDisabled()
+     * - disable()
+     * - enable()
+     *
+     * The default behavior for jobs that do not override these methods is isDisabled() always returns false,
+     * and enable()/disable() should not be called (they throw exceptions). enable()/disable() should only be
+     * called if supportsDisable() returns true.
+     *
+     * @since TODO
+     */
+    public boolean supportsDisable() {
+        return false;
+    }
+
+    /**
+     * Returns true if the job is currently disabled. Returns false if the job is not disabled, or does not
+     * support being disabled.
+     *
+     * @since TODO
+     */
+    public boolean isDisabled() {
+        return false;
+    }
+
+    /**
+     * Subclasses should implement this class if they support enable/disable semantics. Callers should check
+     * supportsDisable() before calling this method.
+     *
+     * Transitions this job to the disabled state. If the job is already disabled this method silently does nothing.
+     *
+     * @since TODO
+     */
+    public void disable() throws IOException {
+        throw new IllegalStateException("does not support disable/enable");
```

Comment:
This is a normal use case for `UnsupportedOperationException` (or at a push, `IllegalStateException`), both of which are unchecked exceptions, so that the method signature contains `IOException` should be irrelevant. &lt;/bikeshed>